### ISSUE TITLE
Balance error

### ISSUE THREAD

**Issue by Jokersonchik** [iron-fish/ironfish#1728]
### What happened?

After checking the balance - there is on the account.
When trying to make a deposit, it gives an error.
When checking the balance again, other values are displayed.
![Balance error](https://user-images.githubusercontent.com/106005810/177346649-cb27b80e-e586-4e06-a7bb-d7a193307ebe.png)

### Version

0.1.37 @ src

### Debug output

```shell
root@Jokersonchiknode:~# ironfish debug
yarn run v1.22.19
$ yarn build && yarn start:js debug
$ tsc -b
$ cross-env OCLIF_TS_NODE=0 IRONFISH_DEBUG=1 node bin/run debug
Database in use, skipping output that requires database.
Stop the node and run the debug command again to show full output.

Iron Fish version       0.1.37 @ src
Iron Fish library       0.0.15 @ src
Operating system        Linux x64
CPU model(s)            Common KVM processor
CPU threads             5
RAM total               4.81 GiB
Node version            v16.15.1
ironfish in PATH        true
Telemetry enabled       true
Done in 2.67s.
```

### Relevant log output

```shell
root@Jokersonchiknode:~# ironfish accounts:balance $IRONFISH_WALLET
yarn run v1.22.19
$ yarn build && yarn start:js accounts:balance Jokersonchiknode
$ tsc -b
$ cross-env OCLIF_TS_NODE=0 IRONFISH_DEBUG=1 node bin/run accounts:balance Jokersonchiknode
Account - Jokersonchiknode (401d350)

The balance is: $IRON 4.93905402 ($ORE 493,905,402)
Amount available to spend: $IRON 2.29468702 ($ORE 229,468,702)
Done in 61.22s.
root@Jokersonchiknode:~# ironfish deposit --confirm
yarn run v1.22.19
$ yarn build && yarn start:js deposit --confirm
$ tsc -b
$ cross-env OCLIF_TS_NODE=0 IRONFISH_DEBUG=1 node bin/run deposit --confirm
Creating the transaction: [████████████████████████████████████████] 100% | ETA: 0s
An error occurred while sending the transaction.
 ›   Error: Request failed (400) error: Insufficient funds
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
root@Jokersonchiknode:~# ironfish accounts:balance $IRONFISH_WALLET
yarn run v1.22.19
$ yarn build && yarn start:js accounts:balance Jokersonchiknode
$ tsc -b
$ cross-env OCLIF_TS_NODE=0 IRONFISH_DEBUG=1 node bin/run accounts:balance Jokersonchiknode
Account - Jokersonchiknode (401d350)

The balance is: $IRON 2.64436700 ($ORE 264,436,700)
Amount available to spend: $IRON 0.00000000 ($ORE 0)
Done in 64.02s.
```

### Graffiti

Jokersonchiknode

**Comment by Jokersonchik** [comment]
I decided to see what would happen after a while, now I tried to do it again:

root@Jokersonchiknode:~# ironfish accounts:balance $IRONFISH_WALLET
yarn run v1.22.19
$ yarn build && yarn start:js accounts:balance Jokersonchiknode
$ tsc -b
$ cross-env OCLIF_TS_NODE=0 IRONFISH_DEBUG=1 node bin/run accounts:balance Jokersonchiknode
Account - Jokersonchiknode (401d350)

The balance is: $IRON 4.75048694 ($ORE 475,048,694)
Amount available to spend: $IRON 2.10136510 ($ORE 210,136,510)
Done in 127.16s.
root@Jokersonchiknode:~# ironfish deposit --confirm
yarn run v1.22.19
$ yarn build && yarn start:js deposit --confirm
$ tsc -b
$ cross-env OCLIF_TS_NODE=0 IRONFISH_DEBUG=1 node bin/run deposit --confirm
Creating the transaction: [████████████████████████████████████████] 100% | ETA: 0s
An error occurred while sending the transaction.
 ›   Error: Request failed (400) error: Insufficient funds
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
root@Jokersonchiknode:~# ironfish accounts:balance $IRONFISH_WALLET
yarn run v1.22.19
$ yarn build && yarn start:js accounts:balance Jokersonchiknode
$ tsc -b
$ cross-env OCLIF_TS_NODE=0 IRONFISH_DEBUG=1 node bin/run accounts:balance Jokersonchiknode
Account - Jokersonchiknode (401d350)

The balance is: $IRON 2.64912184 ($ORE 264,912,184)
Amount available to spend: $IRON 0.00000000 ($ORE 0)
Done in 153.02s.

**Comment by NullSoldier** [comment]
This happened because your node wass being re-orged while you were trying to spend notes. I think your balance changed in the middle of creating the transaction.
